doc/dev/deduplication.rst

```diff
@@ -149,3 +149,64 @@ Aside from radosgw, completing work on manifest object support in the
 OSD particularly as it relates to snapshots would be the next step for
 rbd and cephfs workloads.
 
+How to use deduplication
+========================
+
+Currently, Ceph provides deduplication using RADOS machinery.
+The below is explanation about how users use deduplication.
+
+
+1. Find dedupable chunks using ceph-dedup-tool. ::
+
+    ceph-dedup-tool --op estimate --pool $POOL --chunk-size chunk_size  
+      --chunk-algorithm fixed|fastcdc --fingerprint-algorithm sha1|sha256|sha512
+
+  This CLI command will show how many storage space can be saved when deduplication
+  is applied on the pool. If the amount of the saved space is higher than user's expectation,
+  the pool probably is worth performing deduplication. 
+  Users should specify $POOL where the object---the users want to perform
+  deduplication---is stored. The users also need to run ceph-dedup-tool multiple time
+  with varying ``chunk_size`` to find the optimal chunk size. Note that the
+  optimal value probably differs in the content of each object in case of fastcdc
+  chunk algorithm (not fixed).
+
+
+2. Run dedup command (there are two ways). ::
+  
+    ceph-dedup-tool --op object-dedup --pool POOL --object OID --chunk-pool POOL
+      --fingerprint-algorithm FP --dedup-cdc-chunk-size CHUNK_SIZE
+
+  object-dedup command triggers to perform deduplication on the object specified by OID.
+  Users need to use all parameter above---FP and CHUNK_SIZE---based on the result from step 1.
+  After this command is done, note that fastcdc will be set by default and other parameters 
+  such as FP and CHUNK_SIZE also will be set as the POOL's default parameter.
+  Deduplicated objects will appear in the chunk pool.
+
+  ::
+    ceph-dedup-tool --op chunk-dedup --pool POOL --object OID --chunk-pool POOL
+      --fingerprint-algorithm FP --source-off OFFSET --source-length LENGTH 
+
+  chunk-dedup command runs deduplication on a part of the object---the range is set by OFFSET
+  and LENGTH. 
```

Comment:
Under what conditions would a user actually do this?  Is this a useful option?

Reply:
I hadn't realized that we had dedup at all yet, it's a long-desired feature. One thing I wondered about was that this seems to operate on per-RADOS-object granularity, so one might have to script up a loop with many iterations to achieve significant space reduction.  And since this is after-the-fact vs on-the-fly, it incurs additional writes to media, which on HDDs could degrade performance and on SSDs would represent additional wear.  These points might be noted somewhere in the document.

Reply:
https://github.com/ceph/ceph/pull/45369 afaict is intended to add the background task portion of the feature.  The point of adding documentation at this stage is less that it's ready for use (it really doesn't seem to be) and more to clarify what actually exists, guide development, and enable early testing/feedback.